tests/posix/pthread_pressure/src/main.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <pthread.h>
+
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/ztest.h>
+
+#define STACK_SIZE K_THREAD_STACK_LEN(CONFIG_TEST_STACK_SIZE)
+
+/* 32 threads is mainly a limitation of find_lsb_set() */
+#define NUM_THREADS MIN(32, MIN(CONFIG_TEST_NUM_CPUS, CONFIG_MAX_PTHREAD_COUNT))
+
+LOG_MODULE_REGISTER(pthread_pressure, LOG_LEVEL_INF);
+
+/* bitmask of available threads */
+static atomic_t free_threads =
+	ATOMIC_INIT((NUM_THREADS == 32) ? UINT32_MAX : BIT_MASK(NUM_THREADS));
```

Comment:
As I can see we can simplify it to:

```suggestion
static atomic_t free_threads = ATOMIC_INIT(BIT_MASK(NUM_THREADS);
```

Reply:
Fixed! Thank you